tests/integration/test_execution.py

```diff
@@ -586,3 +623,29 @@ def _test_callback_function_with_another_log_message(payload: Payload):
     logger = logging.getLogger(__name__)
     logger.info("%s is the message that I'll process!", payload.body)
     payload.ack()
+
+
+def _test_send_message_without_correlation_id_header(body: str, queue: str, attempt=1):
+    publisher = build_publisher()
+    standard_header = {
+        "tshoot-destination": queue,
+        # RabbitMQ
+        # These two parameters must be set on consumer side as well, otherwise you'll get precondition_failed
+        "x-dead-letter-routing-key": create_dlq_destination_from_another_destination(queue),
+        "x-dead-letter-exchange": "",
+    }
+
+    send_params = {
+        "destination": queue,
+        "body": json.dumps(body, cls=DjangoJSONEncoder),
+        "headers": standard_header,
+        "content_type": publisher._default_content_type,
+        "transaction": getattr(publisher, "_tmp_transaction_id", None),
+    }
+    send_params = clean_dict_with_falsy_or_strange_values(send_params)
+
+    def _internal_send_logic():
+        publisher.start_if_not_open()
+        publisher.connection.send(**send_params)
+
+    publisher._retry_send(_internal_send_logic, attempt=attempt)
```

Comment:
We didn't make it easy to test, right? Hahahah

I'm not a fan of calling "private" parts of modules or classes whenever you need, that's why I would prefer to mock the `clean_dict_with_falsy_or_strange_values` to return the exactly `send_params` you want and then call the send as is.

Reply:
A lot of time ago... rsrs
I am trying to think in a way to test it without allow devs to send messages without correlation-id. This lib avoid devs to do it (for best practices).
For this reason I can't change send method of Publisher class. So my concern is that retry_send method should be a retry helper method and not a protected method of the class, right?

Reply:
I like that you removed the retry method from the `Publisher` class. It isn't something that concerns only to him.

But, since you're attempting to send once, the retry isn't important here. So, you could call the `publisher.connection.send` directly.